policies/broken-nightlies.md

```diff
@@ -0,0 +1,36 @@
+# Policy on broken nightlies
+
+Sometimes the nightlies released automatically by our CI ends up being broken
+for some people or even everyone. This policy defines what the infra team
+response will be in those cases.
+
+## Which nightly will be rolled back
+
+A nightly can only be rolled back in the following cases:
+
+* If it contains destructive code, for example if the included compiler deletes
+  all the users files.
+* If an infra problem caused it to be broken for a big percentage of users on
+  any Tier 1 platform. Issues affecting only lower tier platforms are not
+  worthy of a roll back, since we don't guarantee working builds for those
+  platforms anyway.
+
+A nightly will **not** be rolled back if it's broken by a critical compiler
+bug: those bugs are supposed to be caught by CI, and nightly can have compiler
+regressions anyway. There are no exceptions, even if big projects are broken
+because of this.
```

Comment:
What's the rationale behind this?

Reply:
Mostly because "rolling back because of a compiler bug" is a big Pandora's box: what's a compiler bug and when is it so critical that it requires a nightly roll back? Most of the really big ones are caught by CI anyway.

Instead this policy allows us to roll back on the rare occasion of an infra failure, which happens rarely but it happens (like the broken manifests). Those are also pretty clear-cut, either everyone is broken or very few people are.